PublishOpenDebugAD7.sh

```
@@ -0,0 +1,131 @@
+#!/bin/sh
+
+# Copyright (c) Microsoft. All rights reserved.
+
+# Working dirctory to return to
+__InitialCWD=$(pwd)
+
+# Location of the script
+__ScriptDirectory=
+
+# RuntimeID to publish
+__RuntimeID=
+
+# Configuration of build
+__Configuration=Debug
+
+# OutputFolder
+__OutputFolder=
+
+log_and_exec_cmd() {
+    echo "##[command] $1"
+    $1
+}
+
+print_help()
+{
+    echo 'PublishOpenDebugAD7.sh [-h] [-c C] [-r R]'
+    echo ''
+    echo 'This script publishes OpenDebugAD7'
+    echo '-h    Prints usage information.'
+    echo '-c C  The configuration to publish OpenDebugAD7. Defaults to Debug.'
+    echo '-r R  The RuntimeID to publish OpenDebugAD7. Defaults to machine arch.'
+    echo '-o O  Folder to output the publish. Defaults to the `dotnet publish` folder.'
+}
+
+# Gets the script directory
+get_script_directory()
+{
+    scriptDirectory=$(dirname "$0")
+    cd "$scriptDirectory" || fail "Command failed: 'cd \"$scriptDirectory\"'"
+    __ScriptDirectory=$(pwd)
+    cd "$__InitialCWD" || fail "Command failed: 'cd \"$__InitialCWD\"'"
+}
+
+# Parses and populates the arguments
+parse_and_get_arguments()
+{
+    while getopts "hc:r:o:" opt; do
+        case $opt in
+            c)
+                __Configuration=$OPTARG
+                ;;
+            r)
+                __RuntimeID=$OPTARG
+                ;;
+            o)
+                __OutputFolder=$OPTARG
+                ;;
+            h)
+                print_help
+                exit 1
+                ;;
+            :)
+                echo "ERROR: Option expected for -$OPTARG"
+                print_help
+                exit 1
+                ;;
+        esac
```

>esac [](http://example.com/codeflow?start=8&length=4)

Should we add a `*)` case to fail if it is something unknown?